sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -1236,6 +1267,62 @@ static void _handle_mtuo(gnrc_netif_t *netif, const icmpv6_hdr_t *icmpv6,
     }
 }
 
+#if GNRC_IPV6_NIB_CONF_DNS
+static uint32_t _handle_rdnsso(gnrc_netif_t *netif, const icmpv6_hdr_t *icmpv6,
+                               const ndp_opt_rdnss_t *rdnsso)
+{
+    uint32_t ltime = UINT32_MAX;
+    const ipv6_addr_t *addr;
+
+    if ((rdnsso->len < NDP_OPT_RDNSS_MIN_LEN) ||
+        (icmpv6->type != ICMPV6_RTR_ADV)) {
+        return ltime;
+    }
+    /* select first if unassigned, search possible address otherwise */
+    addr = (sock_dns_server.port == 0) ? &rdnsso->addrs[0] : NULL;
+    if (addr == NULL) {
+        unsigned addrs_num = (rdnsso->len - 1) / 2;
+        for (unsigned i = 0; i < addrs_num; i++) {
+            if (memcmp(sock_dns_server.addr.ipv6,
+                       &rdnsso->addrs[i],
+                       sizeof(rdnsso->addrs[i])) == 0) {
+                addr = &rdnsso->addrs[i];
+                break;
+            }
+        }
+    }
+#if SOCK_HAS_IPV6
```

Comment:
does this make sense, I would assume if `GNRC_IPV6_NIB_CONF_DNS` is true, we have IPv6 support?

Reply:
IPv6-support for `sock` is independent of GNRC's inclusion of GNRC. So it makes sense.

Reply:
from `sys/Makefile.include`:

```
ifneq (,$(filter gnrc_sock,$(USEMODULE)))
  USEMODULE_INCLUDES += $(RIOTBASE)/sys/net/gnrc/sock/include
  ifneq (,$(filter gnrc_ipv6,$(USEMODULE)))
    CFLAGS += -DSOCK_HAS_IPV6
  endif
endif
```

Hence, I would (recursively) resolve/deduce `GNRC_IPV6_NIB_CONF_DNS` -> `GNRC_IPV6` -> `SOCK_HAS_IPV6`.

Reply:
I'd rather prefer to keep the dependency explicit as it is instead of implicit, which addressing this change would make it.